src/probability_theory/density.lean

```diff
@@ -0,0 +1,339 @@
+where `μ` is a measure on the codomain of `X`. -/
+lemma integral_mul_eq_integral' [is_finite_measure ℙ]
+  {X : α → E} [has_pdf X ℙ μ] {f : E → ℝ} (hf : measurable f) :
+  ∫ x, f x * (pdf X ℙ μ x).to_real ∂μ = ∫ x, f (X x) ∂ℙ :=
+begin
+  by_cases hpdf : integrable (λ x, f x * (pdf X ℙ μ x).to_real) μ,
+  { rw [← integral_map (has_pdf.measurable X ℙ μ) hf.ae_measurable,
+        map_eq_with_density_pdf X ℙ μ,
+        integral_eq_lintegral_pos_part_sub_lintegral_neg_part hpdf,
+        integral_eq_lintegral_pos_part_sub_lintegral_neg_part,
+        lintegral_with_density_eq_lintegral_mul _ (measurable_pdf X ℙ μ) hf.neg.ennreal_of_real,
+        lintegral_with_density_eq_lintegral_mul _ (measurable_pdf X ℙ μ) hf.ennreal_of_real],
+    { congr' 2,
+      { have : ∀ x, ennreal.of_real (f x * (pdf X ℙ μ x).to_real) =
+          ennreal.of_real (pdf X ℙ μ x).to_real * ennreal.of_real (f x),
+        { intro x,
+          rw [mul_comm, ennreal.of_real_mul ennreal.to_real_nonneg] },
+        simp_rw [this],
+        exact lintegral_congr_ae (filter.eventually_eq.mul of_real_to_real_ae_eq (ae_eq_refl _)) },
+      { have : ∀ x, ennreal.of_real (- (f x * (pdf X ℙ μ x).to_real)) =
+          ennreal.of_real (pdf X ℙ μ x).to_real * ennreal.of_real (-f x),
+        { intro x,
+          rw [neg_mul_eq_neg_mul, mul_comm, ennreal.of_real_mul ennreal.to_real_nonneg] },
+        simp_rw [this],
+        exact lintegral_congr_ae (filter.eventually_eq.mul of_real_to_real_ae_eq
+          (ae_eq_refl _)) } },
+    { refine ⟨hf.ae_measurable, _⟩,
+      rw [has_finite_integral, lintegral_with_density_eq_lintegral_mul _
+            (measurable_pdf _ _ _) hf.nnnorm.coe_nnreal_ennreal],
+      have : (λ x, (pdf X ℙ μ * λ x, ↑∥f x∥₊) x) =ᵐ[μ] (λ x, ∥f x * (pdf X ℙ μ x).to_real∥₊),
+      { simp_rw [← smul_eq_mul, nnnorm_smul, ennreal.coe_mul],
+        rw [smul_eq_mul, mul_comm],
+        refine filter.eventually_eq.mul (ae_eq_refl _) (ae_eq_trans of_real_to_real_ae_eq.symm _),
+        convert ae_eq_refl _,
+        ext1 x,
+        exact real.ennnorm_eq_of_real ennreal.to_real_nonneg },
+      rw lintegral_congr_ae this,
+      exact hpdf.2 } },
+  { rw [integral_undef hpdf, integral_undef],
+    rwa ← integrable_iff_integrable_mul_pdf hf at hpdf,
+    all_goals { apply_instance } }
+end
+
+/-- A random variable that `has_pdf` is quasi-measure preserving. -/
+lemma to_quasi_measure_preserving {X : α → E} [has_pdf X ℙ μ] :
+  quasi_measure_preserving X ℙ μ :=
+{ measurable := has_pdf.measurable X ℙ μ,
+  absolutely_continuous :=
+  begin
+    rw map_eq_with_density_pdf X ℙ μ,
+    exact with_density_absolutely_continuous _ _,
+  end }
+
+lemma map_absolutely_continuous {X : α → E} [has_pdf X ℙ μ] :
+  map X ℙ ≪ μ :=
+to_quasi_measure_preserving.absolutely_continuous
+
+lemma have_lebesgue_decomposition_of_has_pdf {X : α → E} [hX' : has_pdf X ℙ μ] :
+  (map X ℙ).have_lebesgue_decomposition μ :=
```

Comment:
Hmm, this is not in scope for this PR, but how did this get called `have_lebesgue_decomposition` instead of `has_lebesgue_decomposition`?

Reply:
I named it `have_lebesgue_decomposition` since I would read \mu and \nu have Lebesgue decomposition